mdx/guides/0x-extensions-explained.mdx

```diff
@@ -14,20 +17,26 @@ Users send ETH and a set of orders to the Forwarder contract and these funds are
 
 The Forwarder extends the behavior of trading on 0x. It presents a different interface where the user can specify the amount of tokens they wish to buy and use ETH to buy them. This is an improved user experience to interacting with the lower level 0x Exchange contract. From a user's perspective it appears as if they are buying tokens using ETH directly.
 
-<img width="575" src="https://user-images.githubusercontent.com/27389/53459986-8dd25800-3a09-11e9-87ce-14eff3464b28.png" />
+<img
+    width="575"
+    src="https://user-images.githubusercontent.com/27389/53459986-8dd25800-3a09-11e9-87ce-14eff3464b28.png"
+/>
 
 Projects can use the Forwarder contract and a Relayers orders in their own integrations. We've seen great adoption of 0x Instant, which uses the Forwarder under the hood. These projects are able to use the affiliate fee in the Forwarder to collect fees in ETH for the trades they originate.
 
 ```typescript
-const txHash = await contractWrappers.forwarder.marketBuyOrdersWithEthAsync(
-    signedOrders,
-    assetAmount,
-    taker,
-    amountInEth,
-    feeOrders,
-    affiliateFeePercentage,
-    affiliateFeeAddress,
-);
+const txHash = await contractWrappers.forwarder
+    .marketBuyOrdersWithEth(
+        [signedOrder],
+        order.makerAssetAmount,
+        [signedOrder.signature],
+        affiliateFee,
+        affiliateFeeRecipient,
+    )
+    .sendTransactionAsync({
+        from: taker,
+        value: order.takerAssetAmount.plus(calculateProtocolFee([signedOrder])),
```

Comment:
Do we expose a `calculateProtocolFee`?

Reply:
We do in 0x-starter-project where these examples are sourced from. Asset-swapper also has a helper. We should write a full guide on protocol fees and calculation imo.